### ISSUE TITLE
[Question] Best practice for targeting an element outside of the current component using hooks?

### ISSUE THREAD

**Issue by ardiewen** [thebuilder/react-intersection-observer#480]
Hi there, first of all, thanks for making and maintaining this super useful library.

Just had a question regarding how to best observe an element outside of the current component using hooks.

### The Use Case

In a contrived example, Let's say I have a deeply nested child component that wants to observe the visibility of the site footer. The DOM tree would look something like this:

```
...
return (
  <header />
  <main>
    <div>
      <MyComponent />
      {/* more components here */}
    </div>
  </main>
  <Footer id="footer" />
)
```

### Possible Solutions

1. `useInView` returns both `ref` and `inView`, so the only way to ref the Footer component and pass inView to MyComponent is to use the hook in a parent component and pass the values down through composition or context API. For simple examples, composition may be enough:

```
function MyApp() {
  const { ref, inView } = useInView()

  return (
    <header />
    <main>
      <div>
        <MyComponent footerInView={inView}/>
        {/* more components here */}
      </div>
    </main>
    <Footer inViewRef={ref} />
  )
}
```

2. Utilizing "target" with the low level API to contain all the logic in `MyComponent`. Something like:

```
function MyComponent() {
   const [inView, setInView] = useState(false)

  useEffect(() => {
    const footerEl = typeof document !== 'undefined' && document.querySelector('#footer')
    const destroy = observe(footerEl, state => setInView(state), { rootMargin: 0 })

    return () => destroy()
  }, [])

  return (
    <div>{inView.toString()}</div>
  )
}
```

3. Is there a possibility to still use the already written hook `useInView` this way instead of the low level `observe` API? Something like this seems to work:

```
function MyComponent() {
  const { ref, inView } = useInView()

  useEffect(() => {
    const footerEl =
      typeof document !== "undefined" && document.querySelector("#footer")

    ref(footerEl)
  }, [ref])

  return (
    <div>{inView.toString()}</div>
  )
}
```

I'm not sure if option 3 above has some unintended consequences, but it does seem to do the trick. Any guidance would be greatly appreciated!

**Comment by thebuilder** [comment]
Option 3 should work fine. 
But it might be better to consider either creating a custom Context that can report the state of the footer/ui to components that needs it, or using a state library like https://github.com/pmndrs/jotai

**Comment by ardiewen** [comment]
Thanks @thebuilder!

**Comment by dspacejs** [comment]
I'd also really like a simple solution for this.

I'm building a sticky navigation for a single page. When a given section is in view, its respective navigation link changes its style to indicate it's active.

However, I can't really use the hooks from this library, without passing them down to many children (which is messy). A rudimentary problem like this should be solvable with a few lines of code at most IMO. Using the native intersection observer API is also verbose (I'm not a fan).

So it would be great if this library was adapted for this problem somehow. @ardiewen I'm personally not a fan of solution 3 because it's verbose. Instead, I imagined something like this:

```tsx
const MyComponent = () => {
  const { inView } = useInView("idOfElement");

  return (
    <div>{inView.toString()}</div>
  );
};
```

And perhaps this could also be adapted somehow to accept an array of elements to track (not just one), e.g with the navigation example I gave.
